modules/core/src/main/java/org/apache/ignite/internal/maintenance/MaintenanceProcessor.java

```diff
@@ -90,28 +91,70 @@ public MaintenanceProcessor(GridKernalContext ctx) {
     }
 
     /** {@inheritDoc} */
-    @Override public @Nullable MaintenanceTask registerMaintenanceTask(MaintenanceTask task) throws IgniteCheckedException {
+    @Nullable @Override public MaintenanceTask registerMaintenanceTask(MaintenanceTask task) throws IgniteCheckedException {
+        return registerMaintenanceTask0(task, null);
+    }
+
+    /** {@inheritDoc} */
+    @Nullable @Override public MaintenanceTask registerMaintenanceTask(
```

Comment:
The method always returns **null**, it doesn't seem to match the signature, I think it should return the old task it replaces with, WDYT?